bin/e2e-test-client/src/test_context.rs

```diff
@@ -186,6 +192,87 @@ impl Wallet {
         Ok(tx.finalize_as_transaction())
     }
 
+    /// Creates the script transaction that collects fee.
+    pub async fn collect_fee_tx(
+        &self,
+        coinbase_contract: ContractId,
+        asset_id: AssetId,
+    ) -> anyhow::Result<Transaction> {
+        // select coins
+        let coins = &self
+            .client
+            .coins_to_spend(
+                &self.address,
+                vec![(AssetId::BASE, BASE_AMOUNT, None)],
+                None,
+            )
+            .await?[0];
+
+        let output_index = 2u64;
+        let call_struct_register = 0x10;
+        // Now call the fee collection contract to withdraw the fees
+        let script = vec![
+            // Point to the call structure
+            op::gtf_args(call_struct_register, 0x00, GTFArgs::ScriptData),
+            op::addi(
+                call_struct_register,
+                call_struct_register,
+                (asset_id.size() + output_index.size()) as u16,
+            ),
+            op::call(call_struct_register, RegId::ZERO, RegId::ZERO, RegId::CGAS),
+            op::ret(RegId::ONE),
+        ];
+
+        // build transaction
+        let mut tx = TransactionBuilder::script(
+            script.into_iter().collect(),
+            asset_id
+                .to_bytes()
+                .into_iter()
+                .chain(output_index.to_bytes().into_iter())
+                .chain(coinbase_contract.to_bytes().into_iter())
+                .chain(0u64.to_bytes().into_iter())
+                .chain(0u64.to_bytes().into_iter())
+                .collect(),
```

Comment:
This doesn't seem match the fee collection contract from https://github.com/FuelLabs/fuel-core/pull/1501, is that on purpose?

Reply:
I've change the order of input arguments https://github.com/FuelLabs/fuel-core/pull/1501/commits/ed06e89f095d54fe3cfaa9753f4a9dc9bf2c48ef

Reply:
Nevermind, that seems right